# topics & frames
target_topic    : "/target_point/pose"
odom_topic      : "/odom"              # defalut： /odom
desirePos_topic : "/desired_pose/pose"        # optional
marker_topic    : "/visualization_marker"     # outdoor test target topic

global_frame    : "odom"
car_frame       : "base_footprint"

# ControlLoop params
thread_numbers  : 2
pub_twist_cmd   : true
debug_info      : false
max_speed       : 1.5
controller_freq : 10
lenth_hand      : 0.5

# Parameter for MPC solver
target_distance : 1.0
target_alpha    : 0
mpc_steps       : 20.0
mpc_max_angvel  : 1.5
# mpc_max_throttle: 1.0
mpc_bound_value : 1.0e6

# Wight params
mpc_w_ex        : 80000.0           # 纵向误差
mpc_w_etheta    : 8000.0            # 航向误差
mpc_w_vel       : 10.0              # 目标速度误差
mpc_w_vel_d     : 100.0             # 速度变化惩罚

mpc_w_angvel    : 10.0            # 角速度惩罚
mpc_w_angvel_d  : 1000.0         # 角速度变化惩罚